apps/authentication/api/views.py

```diff
@@ -83,6 +85,44 @@ def dump_data(self, request, pk: int):
         serializer = self.get_serializer(user)
         return Response(data=serializer.data, status=status.HTTP_200_OK)
 
+    @action(detail=True, methods=["get"], url_path="permissions")
+    def user_permissions(self, request, pk):
+        """
+        Returns all permissions for a given user.
+        Codename is "action_object", e.g. "delete_profile".
+        Content_type is ID of object type.
+        If you want to group permissions, you can group on object type and map the ID to the object from the codename string.
+        """
+        user = self.get_object()
+        permissions = (
+            Permission.objects.filter(group__user=user)
+            .order_by("content_type")
+            .distinct()
+        )
+        page = self.paginate_queryset(permissions)
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+        serializer = self.get_serializer(permissions, many=True)
```

Comment:
Could this lead to the response some times including the pagination object (with total count, results array, next page and so on)?
That may trip over a client that always expects the result to be wrapped in that object

Reply:
That is true! I just figured pagination was a smart thing to keep using when possible since most of the other endpoints does. However, I do agree that you usually want all permissions when you query this endpoint instead of repeated queries until you have reached all pages.

Reply:
I agree that it could be nice to keep using. How is the `None` page handled for regular ModelViewSets? Is it possible to replicate how they do it, like return an empty paginated response or whatever they do?

Reply:
I'm not sure that ever is a thing to be honest. I've tried with 0 data, but it still becomes a QuerySet with 0 data in the results-section. Just found that error-handling at StackOverflow and thought it might be a thing. But I have never been able to trigger a None-page as of yet.